src/makeCompleteAction.js

```diff
@@ -69,18 +69,52 @@ const mountRequest = (name, options) => (requestType, payload) => {
     throw new Error("The request type name should be a string.");
 
   const typesOfActions = {
-    request: () => ({
-      type: mountConstant(name, "request", options),
-      payload
-    }),
-    success: () => ({
-      type: mountConstant(name, "success", options),
-      payload
-    }),
-    fail: () => ({
-      type: mountConstant(name, "fail", options),
-      payload
-    })
+    request: () => {
+      if (options && options.customSuccessPayload) {
+        return Object.assign(
+          {
+            type: mountConstant(name, "request", options)
+          },
+          payload
+        );
+      }
+
+      return {
+        type: mountConstant(name, "request", options),
+        payload
+      };
+    },
+
+    success: () => {
+      if (options && options.customSuccessPayload) {
+        return Object.assign(
+          {
+            type: mountConstant(name, "success", options)
+          },
+          payload
+        );
+      }
+
+      return {
+        type: mountConstant(name, "success", options),
+        payload
+      };
+    },
+
+    fail: () => {
+      if (options && options.customFailurePayload) {
```

Comment:
I see that you expected the custom payloads to be functions. Be aware that a error should be thrown when the user tries to pass another type and this situation should be tested in the unit tests!

Reply:
fixed